services/oauth/v2/destination_info_test.go

```diff
@@ -109,6 +111,145 @@ var isOAuthDestTestCases = []destInfoTestCase{
 			isOAuth: false,
 		},
 	},
+	// scenarios considering only account is present mostly possible for any new oauth destinations
+	{
+		description:    "should return 'true' without error for a OAuth destination when account is presence and refreshOAuthToken is true",
```

Comment:
Add test case where account is present but no definition is present (legacy oauth account) it should fallback to destination definition